stellargraph/mapper/sequences.py

```diff
@@ -530,6 +533,126 @@ def __getitem__(self, index):
         return self.inputs, self.targets
 
 
+class GraphSequence(Sequence):
+    """
+    A Keras-compatible data generator for training and evaluating graph classification models.
+    Use this class with the Keras methods :meth:`keras.Model.fit`,
+        :meth:`keras.Model.evaluate`, and
+        :meth:`keras.Model.predict`,
+
+    This class should be created using the `.flow(...)` method of
+    :class:`GraphGenerator`.
+
+    Args:
+        graphs (list)): The graphs as StellarGraph objects.
+        node_features_size (int): The dimensionality of node features.
+        targets (np.ndarray, optional): An optional array of graph targets of size (N x C),
+            where N is the number of graphs and C is the target size (e.g., number of classes.)
+        normalize (bool, optional): Specifies whether the adjacency matrix for each graph should
+            be normalized or not. The default is True.
+        batch_size (int, optional): The batch size. It defaults to 1.
+        name (str, optional): An optional name for this generator object.
+    """
+
+    def __init__(
+        self,
+        graphs,
+        node_features_size,
+        targets=None,
+        normalize=True,
+        batch_size=1,
+        name=None,
+    ):
+
+        self.name = name
+        self.graphs = np.asanyarray(graphs)
+        self.normalize_adj = normalize
+        self.targets = targets
+        self.batch_size = batch_size
+        self.node_features_size = node_features_size
+
+        if targets is not None:
+            if len(graphs) != len(targets):
+                raise ValueError(
+                    "expected the number of target values and the number of graphs to be the same length,"
+                    f"found {len(graphs)} graphs and {len(targets)} targets."
+                )
+
+            self.targets = np.asanyarray(targets)
+
+        if self.normalize_adj:
+            self.normalized_adjs = [
+                normalize_adj(graph.to_adjacency_matrix()) for graph in graphs
+            ]
+        else:
+            self.normalize_adjs = [graph.to_adjacency_matrix() for graph in graphs]
+
+        self.normalized_adjs = np.asanyarray(self.normalized_adjs)
+
+        self.on_epoch_end()
+
+    def __len__(self):
+        return int(np.ceil(len(self.graphs) / self.batch_size))
+
+    def __getitem__(self, index):
+        graphs = self.graphs[
+            index * self.batch_size : (index * self.batch_size) + self.batch_size
+        ]
+
+        adj_graphs = self.normalized_adjs[
+            index * self.batch_size : (index * self.batch_size) + self.batch_size
+        ]
+
+        # The number of nodes for the largest graph in the batch. We are going to pad with 0 rows and columns
+        # the adjacency and node feature matrices (only the rows in this case) to equal in size the adjacency and
+        # feature matrices of the largest graph.
+        max_nodes = max([graph.number_of_nodes() for graph in graphs])
```

Comment:
I think it is fine as it is. We don't know if some future dataset will a graph that is considerably larger than all others and we might be wasting space to avoid an operation that doesn't seem to be that costly. I don't see any reason to over-optimise at this stage. I want to get some more datasets into the library, some of which are larger than MUTAG, to get a better idea of performance before spending more time on optimisations.